File: DSA/AadrikaBehl_590012053/day6q3.c
#include <stdio.h>

int findLongestConsecutiveSubarray(int nums[], int n) {
    int maxLen = 1, currLen = 1;

    for (int i = 1; i < n; i++) {
        if (nums[i] == nums[i - 1] + 1) {
            currLen++;
            if (currLen > maxLen)
                maxLen = currLen;
        } else {
            currLen = 1; 
        }
    }

    return maxLen;
}

int main() {
    int n;
    printf("Enter number of elements in array: ");
    scanf("%d", &n);

    int nums[n];
    printf("Enter %d integers:\n", n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &nums[i]);
    }

    int result = findLongestConsecutiveSubarray(nums, n);
    printf("Length of longest consecutive increasing subarray: %d\n", result);

    return 0;
}

File: DSA/Muskan_590015043/Day6_Q3.py
# 3. Find the Length of the Longest Consecutive Increasing Subarray
nums = list(map(int,input("Enter the array elements (space seperated) ").split()))
def len_subarray(nums):
    new = []
    for i in range(len(nums)-1,0,-1):
        if nums[i]==nums[i-1]+1:
            new.append(0)
        else:
            new.append(1)
    count = 0
    for j in new:
        if j == 0:
            count = count+1
        elif j == 1:
            break
    return count+1
print(len_subarray(nums))
